portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java

```diff
@@ -148,6 +149,17 @@ public DLFolder addFolder(
 
 		dlFolder = dlFolderPersistence.update(dlFolder);
 
+		// Permission Propagation
+
+		long dlFolderClassNameId = _classNameLocalService.getClassNameId(
+			DLFolder.class);
+		boolean permissionPropagationEnabled = ParamUtil.getBoolean(
+			serviceContext, "permissionPropagationEnabled");
+
+		_permissionPropagationEntryLocalService.addPermissionPropagationEntry(
+			groupId, serviceContext.getCompanyId(), dlFolderClassNameId,
+			dlFolder.getFolderId(), permissionPropagationEnabled);
```

Comment:
Maybe it is should change: 
`
		_permissionPropagationEntryLocalService.addPermissionPropagationEntry(
			groupId, serviceContext.getCompanyId(), _classNameLocalService.getClassNameId(
			DLFolder.class),
			dlFolder.getFolderId(), ParamUtil.getBoolean(
			serviceContext, "permissionPropagationEnabled"));
`

Reply:
Should use user.getCompanyId() instead of serviceContext.getCompanyId() to avoid companyId = 0

Reply:
Updated! Thanks